lib/net/peer.js

```diff
@@ -81,6 +81,9 @@ class Peer extends EventEmitter {
     this.invQueue = [];
     this.onPacket = null;
 
+    this.dosScore = 0;
```

Comment:
What is your reasoning around adding a new property `dosScore` when `banScore` already exists to handle banning peers?

Reply:
The idea is to have a constant score like `bansScore` which increases linearly, and this can continue to be used as usual. However, dos attempts generally need to be distinguised form regular client messages with further inspection usually based on time (eg a regular peer might request 10 `getproof` per minute, whereas a malicious peer might attempt 1000 `getproof` per minute). This new score needs to increase when requests/minute increase but decay and reset to 0 as requests decrease. They are conceptually different but can be added together to get a score indicating both different types of malicious requests as well mutliple attempts at abusing some of these requests.